streams/src/test/java/org/apache/kafka/streams/state/internals/SegmentIteratorTest.java

```diff
@@ -105,6 +106,46 @@ public void shouldIterateOverAllSegments() {
         assertFalse(iterator.hasNext());
     }
 
+    @Test
+    public void shouldNotThrowExceptionOnHasNextWhenStoreClosed() {
+        iterator = new SegmentIterator(Arrays.asList(segmentOne, segmentTwo).iterator(),
+                                       hasNextCondition,
+                                       Bytes.wrap("a".getBytes()),
+                                       Bytes.wrap("z".getBytes()));
+        segmentTwo.close();
+        segmentOne.close();
+
+        iterator.currentIterator = new KeyValueIterator<Bytes, byte[]>() {
```

Comment:
Why do we need this? I thought `close()` will trigger an exception in the inner iterator already?

Reply:
ack. Cleaned the test up.